### ISSUE TITLE
delay node in the rate limit mode works only as 1 msg per a time frame, not as N msg

### ISSUE THREAD

**Issue by AlexBGoode** [node-red/node-red#1477]
Hello,

I inspected the code and got suspicious about the logic, and my tests just confirmed my conclusion that this node in the rate limit mode doesn't work fairly to limit N messages per T (specified time) frame, instead it works to limit 1 message per T/N

Which is not the same thing

The test case:

if you set the rate 2 messages per 1 minute and send 2 messages one by one, then the first message will pass through however the second message will wait 30 seconds (half of minute, 60/2) instead of passing as well. So it works straight as 1 message per 30 seconds.

For example, the limit 10 messages per hour should allow 10 messages to go within first minute and keep 11th message in a buffer 59 minutes.

Don't you think so?

**Comment by knolleary** [comment]
Hi,

it's all in the interpretation. Neither are wrong, it just so happens that the node implements a different rate limiting to what you were expecting.

In every use case we've had for the node, the behaviour as implemented was the required behaviour - in other words to modify the rate the messages pass through the node - not to send a burst of messages through then wait a long period before sending the next burst.

We could consider adding an optional mode to the node that behaves as you describe; its just a question of designing the right way to expose the option to the user and to be clear in how it differs from the default behaviour.

**Comment by dceejay** [comment]
the original design/UI did just have   "1 message per x Seconds"  (or whatever unit)... so it was perhaps slightly clearer... but then users had to do maths to convert 10 messages per minute to be one every 6 seconds... which is fine for simple cases but can get hard, so we changed the UI to try to help, by accepting both ways of expressing it. Either way the intent was to try to eliminate bursts and spread the output more evenly over a time period, so as not to swamp things downstream.

But yes it could be implemented in other ways.

**Comment by knolleary** [comment]
So, the conclusion here is that isn't an issue.

Do we want to add the alternative mode as an option? If not, lets close this.

**Comment by bonjour81** [comment]
Hi there!
Maybe it's quite late, but I think it could be great to have this options.
Current behavior is not obvious.

I planned to use this rate limit as security option (a few days ago I created a kind of loop that made one of my lamps switch ON/OFF very fast)... so I added in my flow a limit like 60times/hours.
I expected this to be invisible in normal use but would stops any unexpected loops quickly.

At least, I would suggest to write somewhere that "10 times per hours" mean "1 times per 6 minutes".

Best regards,
